### DISCUSSION TITLE
Library Feedback for Version 5

### DISCUSSION THREAD

**Discussion by sh30801** [DSharpPlus/DSharpPlus#991] [Ideas]
Hi everyone, 

I opened this issue as a tracker for what the community would like to see changed in v5 for DSharpPlus. We've been waiting a long time to release v4 and we're excited to say that has finally come true 🎉 

Despite this, we know that this new version of the lib certainly has improvements that can be made, which is why we aim to redesign the lib internals and just generally do a rewrite of the codebase for the next version. If you have an idea/suggestion for the rewrite, or just a strategy we should follow when we rewrite it, please reply with it below.

**Comment by CabbageAdi** [comment]
It always felt like having the `DiscordRestClient` in its separate package was a waste. It's very useful for cutting down on api requests when you can just pass in the id instead of making extra api calls to get the object for it first, and should be more accessible. Now that we're working towards a new major release, my proposal is that we delete the `DSharpPlus.Rest` package altogether, add the `DiscordRestClient` to the main package, and add a `Rest` property to the `DiscordClient`, which is a rest client initialized at startup which can be used easily to make direct rest requests.

**Comment by Naamloos** [comment]
I agree on that one, the `DiscordRestClient` was more of a "quick" solution on adding a rest-only package. Considering it still uses the main lib anyway, there's no point to having that abstraction instead of making parts of the Rest client public. Good call! :)

**Comment by Naamloos** [comment]
From what I remember, it essentially only adds public methods for all private methods the Rest client contains.

**Comment by sh30801** [comment]
Closing this as discussion has moved to https://github.com/DSharpPlus/DSharpPlus/discussions/990
